boards/shields/mikroe_wifi_bt_click/doc/index.rst

```diff
@@ -0,0 +1,78 @@
+.. _mikroe_wifi_bt_click:
+
+MikroElektronika WIFI and BLE Shield
+####################################
+
+Overview
+********
+
+This shield contains an Espressif Systems ESP32 which is responsible
+to offload the WIFI and Bluetooth tasks from the host processor. It uses
+the standard MikroBus interface with UART pins plus its Espressif AT
+command set. Alternatively, the Zephyr modem driver can be used,
+configuring the module as a native WIFI networking interface.
+
+.. image:: ./wifi_click.jpeg
+   :width: 634px
+   :align: center
+   :alt: MikroE WIFI BLE Click
+
+Pins Assignment of the Shield
+=============================
+
++-----------------------+---------------------------------------------+
+| Shield Connector Pin  | Function                                    |
++=======================+=============================================+
+| RST#                  | Reset pin, in most of the boards it should  |
+|                       | be disconnected because its tied to GND     |
++-----------------------+---------------------------------------------+
+| TX                    | UART TX communication pin                   |
++-----------------------+---------------------------------------------+
+| RX                    | UART RX communication pin                   |
++-----------------------+---------------------------------------------+
+
+Requirements
+************
+
+This shield requires a board which provides a configuration that allows an
+UART interface. (see :ref:`shields` for more details).
+
+.. note::
+   Sometimes boards declare standard headers like Arduino R3 but don't
+   define all connections.  Make sure that the board you are using includes
+   all required definitions to avoid build errors.
+
+This shield should be loaded with the `ESP32 AT Bin`_ software which is
+available at Espressif Systems web site. This version is command
+compatible with ESP32 AT Bin 2.0.0
```

Comment:
Sorry to nag, but could you perhaps add a comment and code block here showing how people can flash the said firmware.

It takes a bit of digging to find the right command, and it's important to be able to reproduce the same device config used by this PR:

```bash
$ esptool.py \
  --chip auto \
  --baud 115200 \
  --before default_reset \
  --after hard_reset write_flash \
  --flash_mode dio \
  --flash_freq 40m \
  --flash_size detect \
  0x10000 ota_data_initial.bin \
  0x1000 bootloader/bootloader.bin \
  0x20000 at_customize.bin \
  0x21000 customized_partitions/ble_data.bin \
  0x24000 customized_partitions/server_cert.bin \
  0x26000 customized_partitions/server_key.bin \
  0x28000 customized_partitions/server_ca.bin \
  0x2a000 customized_partitions/client_cert.bin \
  0x2c000 customized_partitions/client_key.bin \
  0x2e000 customized_partitions/client_ca.bin \
  0x30000 customized_partitions/factory_param.bin \
  0xf000 phy_init_data.bin 0x100000 esp-at.bin \
  0x8000 partitions_at.bin
```

Reply:
This is also missing the serial port, baud rate, etc., which should be added just for a working example even if it's a bit OS-specific in the case of the serial port.

Reply:
It is possible to use command without those parameters, esptool will search the board automagically, altough it is possible to supply port and baud-rate, I added these section in the docs.